<a href="https://colab.research.google.com/github/Shihab026/Neurofive-ml-track/blob/main/Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Load Dataset and Initial Inspection

I am loading the `train.csv` dataset into a pandas DataFrame and then performing initial inspections using `.head()`, `.info()`, and `.describe()` to understand its structure and content.

In [1]:
import pandas as pd

# Load dataset
df = pd.read_csv("train.csv")

# Inspect data
print(df.head())
print(df.info())
print(df.describe())

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
<c

### Data Characteristics Summary

Based on the initial inspection, here's a summary of the dataset's characteristics:

In [2]:
# Number of rows and columns
num_rows, num_cols = df.shape
print(f"Number of rows: {num_rows}")
print(f"Number of columns: {num_cols}")

# Identify columns with missing values
missing_values = df.isnull().sum()
missing_values = missing_values[missing_values > 0]
print("\nColumns with missing values and their counts:")
if not missing_values.empty:
    display(missing_values)
else:
    print("No columns with missing values.")

# Separate categorical and numerical columns
numerical_cols = df.select_dtypes(include=['int64', 'float64']).columns.tolist()
categorical_cols = df.select_dtypes(include=['object']).columns.tolist()

print("\nNumerical columns:")
print(numerical_cols)

print("\nCategorical columns:")
print(categorical_cols)

Number of rows: 891
Number of columns: 12

Columns with missing values and their counts:


,0
Age,177
Cabin,687
Embarked,2



Numerical columns:
['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']

Categorical columns:
['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked']


The Titanic dataset contains 891 training rows and 418 test rows, with a mix of numerical and categorical features. Several columns have missing values, notably `Age`, `Cabin`, and `Embarked`, with `Cabin` being largely incomplete. Numerical features include variables like `Age`, `Fare`, and `Pclass`, while categorical features include `Sex`, `Embarked`, and `Ticket`. The target variable `Survived` is present only in the training data. Overall, the dataset will require preprocessing, especially handling missing values and encoding categorical variables, before building a machine learning model.